modules/tasks/tasks/src/main/java/org/motechproject/tasks/web/ActivityController.java

```diff
@@ -93,4 +99,16 @@ public long getTaskActivityCount(@PathVariable Long taskId, @PathVariable String
     public void deleteActivitiesForTask(@PathVariable Long taskId) {
         activityService.deleteActivitiesForTask(taskId);
     }
+
+    /**
+     * Retries task execution for activity with the given ID.
+     *
+     * @param activityId the ID of activity for which task should be retried
+     */
+    @RequestMapping(value = "/activity/retry/{activityId}", method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.OK)
+    public void retryTask(@PathVariable Long activityId) {
+        TaskActivity activity = activityService.getTaskActivityById(activityId);
```

Comment:
Can we move this logic to the task trigger handler and make it one transaction there?